awsmp/models.py

```diff
@@ -764,6 +764,51 @@ def _convert_terms_to_dict(self) -> dict[str, Any]:
 
         return yaml_config
 
+    @staticmethod
+    def _mask_unchecked_fields(yaml_config: dict[str, Any]) -> dict[str, Any]:
+        """
+        Insert stub values for fields that are itrnoed during diffing.
+
+        The `diff` command skips the `version` and `eula_docuemnt` keys,
+        but the data mdel still checks those fields. This helper
+        put placeholder data for those keys so the config passes validation
+        without changing the real local config file.
+
+        :param dict[str, Any] yaml_config: dictionary data from loading local yaml config file
+        :return: dictionary data with stub values in unchecked fields
+        :rtype: dict[str, Any]
+        """
+
+        if "version" in yaml_config["product"]:
+            yaml_config["product"].pop("version")
+        if "eula_document" in yaml_config["offer"]:
+            yaml_config["offer"].pop("eula_document")
```

Comment:
it seems like we can completely skip this since we are overwriting the keys anyway.